### PR TITLE
Feature soft-delete

### PR DESCRIPTION
**Pull request by marceldobrica**

### CODE REVIEW

**Comment by marceldobrica** [comment]
Am incercat sa pastrez commitul cat mai legat de feature.... Aveti dreptate daca lucrurile sunt cat mai specifice se vede bine ce am facut..... dar de ... suntem la inceput....

File: src/Controller/UserController.php [hunk 1]
```diff
@@ -80,23 +81,57 @@ public function register(UserDto $userDto): Response
...
+     */
+    public function deleteUserAction($id): Response
+    {
+        $user = $this->userRepository->findUserById($id);
+
+        if (null === $user) {
+            return new JsonResponse('No user found', Response::HTTP_BAD_REQUEST);
+        }
+        $this->userRepository->remove($user);
+        $this->userRepository->add($user);
```
Comment (Raresmldvn) [inline src/Controller/UserController.php]: why do you remove and then add the user? the idea is to remove a user with soft delete enabled

File: src/Controller/UserController.php [hunk 2]
```diff
@@ -80,23 +81,57 @@ public function register(UserDto $userDto): Response
...
+    {
+        $data = $request->getContent();
+        $decodedData = json_decode($data, true);
+        if (!isset($decodedData['email'])) {
+            $this->logger->warning('An atempt to post on recover without email');
+            return new JsonResponse(
+                'You should provide an email',
+                Response::HTTP_BAD_REQUEST
+            );
+        }
```
Comment (Raresmldvn) [inline src/Controller/UserController.php:107]: a newline here. the code is too crowded in this methid

File: src/Controller/UserController.php [hunk 3]
```diff
@@ -80,23 +81,57 @@ public function register(UserDto $userDto): Response
...
+        $this->entityManager->getFilters()->disable('softdeleteable');
+        $user = $this->userRepository->findOneBy(['email' => $decodedData['email']]);
+        if (null === $user) {
+            $this->logger->warning('No user for recover', ['email' => $decodedData['email']]);
+            return new JsonResponse(
+                'No user with provided email',
+                Response::HTTP_BAD_REQUEST
+            );
+        }
```
Comment (Raresmldvn) [inline src/Controller/UserController.php:117]: also newline here

File: src/Controller/UserController.php [hunk 4]
```diff
@@ -80,23 +81,57 @@ public function register(UserDto $userDto): Response
...
+        $user->setDeletedAt(null);
+        $this->userRepository->add($user);
```
Comment (Raresmldvn) [inline src/Controller/UserController.php]: it's very weird that your "update" method on a user is called `add`.. it looks like you create a new user here

File: src/Entity/User.php [hunk 5]
```diff
@@ -78,6 +80,11 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface
      */
     private Uuid $token;
 
+    /**
+     * @ORM\Column(type="datetime", name="deletedAt")
+     */
+    private ?\DateTime $deletedAt;
```
Comment (Raresmldvn) [inline src/Entity/User.php]: your property hint here is `?\DateTime` but the setter/getter use `?\DateTimeInterface`

File: src/Controller/UserController.php (hunk 4 above)
Comment (marceldobrica) [reply src/Controller/UserController.php]: Yes. Same as first observation... a lot of inconsistencies

File: src/Entity/User.php (hunk 5 above)
Comment (marceldobrica) [reply src/Entity/User.php]: Right. I'll make correction. Also maker-bundle... definitively I should verify differences...

File: src/Controller/UserController.php (hunk 2 above)
Comment (marceldobrica) [reply src/Controller/UserController.php:107]: ok

File: src/Controller/UserController.php (hunk 3 above)
Comment (marceldobrica) [reply src/Controller/UserController.php:117]: ok

File: src/Controller/UserController.php (hunk 4 above)
Comment (marceldobrica) [reply src/Controller/UserController.php]: Solved.

File: src/Controller/UserController.php (hunk 1 above)
Comment (marceldobrica) [reply src/Controller/UserController.php]: solved.

**Pull request merged**